reframe/core/launchers/mpi.py

```diff
@@ -22,6 +22,38 @@ def command(self, job):
         return ['ibrun']
 
 
+@register_launcher('upcrun')
+class UpcrunLauncher(JobLauncher):
+    '''Launcher for UPC applications.'''
+
+    def command(self, job):
+        cmd = ['upcrun']
+        if job.num_tasks:
+            if job.num_tasks_per_node:
+                num_nodes = job.num_tasks // job.num_tasks_per_node
+                cmd += ['-N', str(num_nodes)]
+
+            cmd += ['-n', str(job.num_tasks)]
+
+        return cmd
+
+
+@register_launcher('upcxx-run')
+class UpcxxrunLauncher(JobLauncher):
+    '''Launcher for UPC++ applications.'''
+
+    def command(self, job):
+        cmd = ['upcxx-run']
+        if job.num_tasks:
```

Comment:
You don't need this if. It's guaranteed at this point that `job.num_tasks` will be a positive integer.